lib/semantics/resolve-names.cc

```diff
@@ -6272,14 +6273,30 @@ void ResolveNamesVisitor::ResolveExecutionParts(const ProgramTree &node) {
   SetScope(*node.scope());
   if (const auto *exec{node.exec()}) {
     Walk(*exec);
-    OmpAttributeVisitor{context(), *this}.Walk(*exec);
   }
   PopScope();  // converts unclassified entities into objects
   for (const auto &child : node.children()) {
     ResolveExecutionParts(child);
   }
 }
 
+// Resolve names for OpenMP in the specification and execution part of this node
+// and its children
+void ResolveNamesVisitor::ResolveOmpParts(const ProgramTree &node) {
```

Comment:
If you don't need to do all of the specs before all of the exes, you can probably just call OmpAttributeVisitor on the entire parse tree for a program unit and get rid of this function.

Reply:
I've removed `ResolveOmpParts` function and called `OmpAttributeVisitor{context(), *this}.Walk(x);` in `ResolveNamesVisitor::Pre(const parser::ProgramUnit &x)` instead. Thanks for the suggestions.